buda/entities/signup.proto

```diff
@@ -11,6 +11,8 @@ message Signup {
     Uuid user_id = 2;
     Uuid visitor_id = 3;
     Uuid legacy_visitor_id = 100; //buffer-web visitor_id, will be deprecated
-
+    Uuid client_id = 5;
```

Comment:
I wonder if we could perhaps make this a string or enum field instead of an id? Right now we have to join the client_id with another table to get the name, but I think since we have good idea of all the channels that a signup can come from (web, apple and android I think?) we could just make it a bit more explicit.

Let me know what you think @mallen5311. totally possible that I'm missing something!

Reply:
Great question, @michael-erasmus !  Much hunch in this situation is that we might want to stick to using client_id.  This feels a bit more flexible moving forward, as we do have non-buffer developed apps creating accounts (or at least we attribute the account signup to them in the actions taken, not sure if the account is actually created there ;) ).  We also have roughly 3700 users that were created from an unknown client in the last 30 days, which could be coming from more than we have listed.  That said, we do have the vast majority of user accounts created from web, Buffer Android, and Buffer iOS, so if we can perhaps validate that those are the only ones we actually accept valid user accounts created from, then maybe an enum would work well here.  Here is a look that shows the current breakdown: https://looker.buffer.com/explore/users/users?toggle=fil,pik&qid=6LjWlr1E1ATvodcsC9K8kx

Reply:
Hey @mallen5311! Thanks for pointing that out, awesome to look at the data to help inform how we design the tracking API!

I chatted with @djfarrelly about this a little and I think he might leave some thoughts too, but I have a hunch that the unknown clients might just be a bug in our tracking where we don't report the client, and I think that most of them are probably still web signups.

I'm surprised about the tiny number of other clients (Buffer for Blackberry still cracks me up :smile:). I wonder if we should spend a lot of time dealing with a really small number of edge cases here? Perhaps these can just be filed as a `other` signup client?

Eager to hear @djfarrelly's thoughts too!

Reply:
Thanks for sharing that context, @michael-erasmus (and @djfarrelly )! 😄   Happy to follow your lead with what you think might be best here 👍